runtime/signals.c

```diff
@@ -296,14 +297,16 @@ CAMLprim value caml_install_signal_handler(value signal_number, value action)
     res = Val_int(1);
     break;
   case 2:                       /* was Signal_handle */
-    caml_read_field(caml_read_root(caml_signal_handlers), sig, &handler);
+    caml_read_field(caml_signal_handlers, signal_number, &handler);
```

Comment:
I think we might have a typo here. Shouldn't this be:
`caml_read_field(caml_signal_handlers, sig, &handler);`